Automattic/tumblr-theme-garden
# Make theme browsing React-y

### Summary

While I was trying to make a [theme details overlay](https://github.com/Automattic/T3/pull/37), I realized that the theme garden is going to need some more javascript.
I decided to go with React.

In this PR, I refactor the theme garden so that we can use React. There should be no functionality changes.

### Testing steps

Run a smoke test to make sure you can still change categories, use search, and activate themes.
